doh-server/cache.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"encoding/json"
+	"hash/fnv"
+	"log"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+const debugCache = false
```

Comment:
This could be moved to configuration, but it is mainly for debugging the cache feature itself.